pkg/github/client.go

```diff
@@ -22,13 +22,24 @@ type Interface interface {
 	MergeBranches(base, head string) (bool, error)
 }
 
-func New(token, owner, repo string, ctx context.Context) (Interface, error) {
+func New(token, owner, repo, baseURL string, ctx context.Context, isEnterprise bool) (Interface, error) {
 	// Authenticate with Github
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	client := gogithub.NewClient(tc)
+	var client *gogithub.Client
+	var err error
+
+	if isEnterprise {
+		client, err = gogithub.NewEnterpriseClient(baseURL, baseURL, tc)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		client = gogithub.NewClient(tc)
```

Comment:
Is this always creating a client for github.com?

If yes, then you can basically remove the `isEnterprise` flag and just infer which client to initialise based on the `baseURL` (if it is empty -> default, if it is https://github.com -> default, else Enterprise)

Reply:
yes, can do that